.github/workflows/deno.yml

```diff
@@ -10,9 +10,7 @@ name: grammY
 
 on:
     push:
-        branches: [main]
-    pull_request:
-        branches: [main]
+        branches: ["*"]
```

Comment:
Removing `pull_request` means that external pull requests are no longer run against the CI. Its unlikely that this is wanted.

Reply:
Ohhhh I was not aware, thanks for pointing it out. I'll revert this part of the changes

Reply:
I'd expect that the rules collectively determine whether or not CI runs, rather than determining the number of times that identical jobs are started. Are you sure about this?

Reply:
Please see #749

Reply:
Holy crap, you were right. Wtf

Reply:
one runs the branch, one merges the branch into main and then runs the CI. When PR aren't on top of the main this might be different → different meaning → two jobs.

Reply:
@wojpawlik I think I have fixed it now, the trade-off is that jobs will only be duplicated when we merge the v2 branch via a PQ, which only happens once. Can you TAL?

Reply:
> one merges the branch into main and then runs the CI

Nope. Pushing to a PQ that wants to merge A into B triggers `push` on A and `pull_request` on B.

Reply:
I think we are talking past each other. You talk about what triggers them, I talk about what runs and why GitHub doesn’t deduplicate them.

`pull_request` merges with the target branch (in most cases main) before running the CI. (And doesn’t run when a merge can’t be done automatically.) `push` does not do merging. So they have different goals with different meaning.